File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-discover/src/services/discoverService.ts
import type { AppstoreAppClient } from "@sdkwork/appstore-mp-core";

import type { DiscoverPageResult } from "../types/discoverModels.js";

/**
 * Discover service.
 *
 * The generated App Store app SDK client is injected by the root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export interface DiscoverService {
  readonly capability: "discover";
}

export function createDiscoverService(client: AppstoreAppClient): DiscoverService {
  void client;
  return { capability: "discover" };
}

export function emptyDiscoverResult(): DiscoverPageResult<never> {
  return { items: [] };
}

File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_discover/lib/src/services/discover_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/discover_models.dart';

/// Discover service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class DiscoverService {
  const DiscoverService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'discover';

  DiscoverPageResult<Never> empty() =>
      const DiscoverPageResult<Never>(items: <Never>[]);
}
